packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -204,9 +204,9 @@ export function evaluateTemplate(vm: VM, html: Template): VNodes {
     if (process.env.NODE_ENV !== 'production') {
         assert.isTrue(
             isFunction(html),
-            `evaluateTemplate() second argument must be an imported template instead of ${toString(
-                html
-            )}`
+            `Invalid template returned by the render() method on ${vm}. It must return an imported template (e.g.: \`import html from "./${
```

Comment:
The component is actually allowed to override the default template with any LWC-compiled template. This assertion just validates that it's a function. I wonder if we can remove it since we use `isTemplateRegistered()` at a later stage?

Reply:
Put another way: if you do

```js
render() {
  return function isNotATemplate () {}
}
```

... then you will get a different error later on. We should probably have consistent errors for anything that is not a registered template function.